frontend/cypress/support/assertions.js

```diff
@@ -0,0 +1,20 @@
+const isInViewport = (_chai, utils) => {
+    function assertIsInViewport(options) {
+  
+      const subject = this._obj;
+  
+      const bottom = Cypress.$(cy.state('window')).height();
+      const rect = subject[0].getBoundingClientRect();
+  
+      this.assert(
+        rect.top < bottom && rect.bottom < bottom,
+        "expected #{this} to be in viewport",
+        "expected #{this} to not be in viewport",
+        this._obj
+      )
+    }
+  
+    _chai.Assertion.addMethod('inViewport', assertIsInViewport)
+  };
+  
+  chai.use(isInViewport);
```

Comment:
Will look into documentation for this. If you could put this into the relevant tickets you create from this PR.